src/midifile.jl

```diff
@@ -111,6 +168,9 @@ function writeMIDIFile(filename::AbstractString, notes::Notes)
     return midi
 end
 
+@deprecate writeMIDIFile(filename::AbstractString, data::MIDIFile) FileIO.save(filename, data)
```

Comment:
It's more controllable to pass the depredation to 'MIDI.save' instead of 'FileIO.save'. same to 'load'.

Reply:
...? What. The macro is `@deprecate old new [ex=true]`, how can it be that we will deprecate _the new expression_ in favor of the _old_?

Reply:
Sorry for the inacuraccy:

```suggestion
@deprecate writeMIDIFile(filename::AbstractString, data::MIDIFile) save(filename, data)
```

and similar to `load`

Reply:
> (It's FileIO that depends on this package and not this package depends on FileIO.)

This is so confusing. How does FileIO depend on MIDI? Doesn't MIDI.jl just extend the `FileIO.save` method?

Reply:
MIDI creates a new function `save`/`load`, which is not `FileIO.save`/`FileIO.load`. MIDI only tags the `MIDI.load`/`MIDI.save` function with `FileIO.File` type, so that `FileIO.load`/`FileIO.save` calls the `MIDI.load`/`MIDI.save` via `invokelatest`. That's why I said "FileIO depends on MIDI".

Ref: https://juliaio.github.io/FileIO.jl/stable/implementing/

Reply:
```julia
@deprecate writeMIDIFile(filename::AbstractString, data::MIDIFile) save(filename, data)
```
@johnnychen94 MIDI.load and MIDI.save functions take`File{format"MIDI"}` type in its argument and passing the filename string to them gives a method error. FileIO.load and FileIO.save takes the string as input and it works fine with them.

Reply:
You can instead use:

```julia
@deprecate writeMIDIFile(filename::AbstractString, data::MIDIFile) save(File{format"MIDI"}(filename), data)
```

The problem with calling `FileIO.save` is that `FileIO.save` has a try-and-call mechanism to loop the backends, currently, there might only be `MIDI` that supports this file format, but what if there's another package, say `MIDI2`, which happens to have higher priority than `MIDI` in `FileIO`? The answer is, when the user calls `MIDI.writeMIDIFile`, he is eventually calling `MIDI2.save`, which is definitely not what we want.

Reply:
Thanks! Done (in another PR).